onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -6403,10 +6403,23 @@ def aten_pinverse(self: TensorType, rcond: float = 1e-15) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_pixel_shuffle(self: TensorType, upscale_factor: int) -> TensorType:
+@torch_op("aten::pixel_shuffle")
+def aten_pixel_shuffle(self: TReal, upscale_factor: int) -> TReal:
     """pixel_shuffle(Tensor self, int upscale_factor) -> Tensor"""
-
-    raise NotImplementedError()
+    self_shape = op.Shape(self)
+    batch = self_shape[:-3]
+    C_out = op.Unsqueeze(self_shape[-3], [0])
+    H_out = op.Unsqueeze(self_shape[-2], [0])
+    W_out = op.Unsqueeze(self_shape[-1], [0])
+    # Reshaping input by collapsing all leading dimensions to match ONNX op requirement (4D)
```

Comment:
This is a common operation, but ONNX doesn't give us a convenient way of expressing this. We could potentially extend either the [Flatten op](https://onnx.ai/onnx/operators/onnx__Flatten.html#l-onnx-doc-flatten) or [Reshape op](https://onnx.ai/onnx/operators/onnx__Reshape.html) to support this. This is a one-sided flatten. Alternatively, if reshape has an attribute to align the shape with existing shape starting from the end (instead of beginning), we could specify the new target shape as `[-1, 0, 0, 0]`. Just some idle thoughts ...

Reply:
https://github.com/microsoft/onnxscript/issues/1539